File: ruby/raindrops/raindrops.rb
class Raindrops

  def self.convert(n)
    begin
      res = ""
      if n % 3 == 0
        res += "Pling"
      end
      if n % 5 == 0
        res += "Plang"
      end
      if n % 7 == 0
        res += "Plong"
      end
      if res == ""
        res = n.to_s
      end
      return res
    end
  end

end

File: haskell/raindrops/src/Raindrops.hs
module Raindrops (convert) where

convert :: Int -> String
convert n 
  | n `mod` 3 == 0 && n `mod` 5 == 0 && n `mod` 7 == 0 = "PlingPlangPlong"
  | n `mod` 3 == 0 && n `mod` 5 == 0 = "PlingPlang"
  | n `mod` 3 == 0 && n `mod` 7 == 0 = "PlingPlong"
  | n `mod` 5 == 0 && n `mod` 7 == 0 = "PlangPlong"
  | n `mod` 3 == 0 = "Pling"
  | n `mod` 5 == 0 = "Plang"
  | n `mod` 7 == 0 = "Plong"
  | otherwise = show n
